### DISCUSSION TITLE
v0.8.0

### DISCUSSION THREAD

**Discussion by arxanas** [arxanas/git-branchless#1052] [Release notes]
To install or update `git-branchless`, run the following:

```
$ cargo install --locked git-branchless
$ git branchless init  # in your repository
```

If you haven't already taken [the user survey](https://github.com/arxanas/git-branchless/discussions/330), please consider doing so!

## Release highlights

- git-branchless is now dual-licensed as MIT/Apache 2.0.
- Rust v1.67 or later is required to build.
- The partial commit interface (`git record -i`) now uses a new, more performant interface.
  - You can also access the interface directly via [`git branchless difftool`](https://github.com/arxanas/git-branchless/wiki/Command:-git-branchless-difftool).
- git-branchless supports repositories managed with `repo` (e.g. Chromium)

### Added

- (#545) EXPERIMENTAL: Added a `--fixup` option to `git move` to squash moved commits into the destination
- (#830) Added `git branchless install-man-pages` command. This may be useful for package maintainers or those who install git-branchless from source.
- (#840) git-branchless supports Phabricator as a backend forge for `git submit`.
- (#845) Added the `branchless.smartlog.defaultRevset` configuration variable.
- (#910) Added support for repositories managed by the `repo` tool.
- (#1009) Added `git branchless difftool` subcommand.

### Changed

- BREAKING (#841): git-branchless is now dual-licensed as MIT/Apache 2.0.
- BREAKING: (#1024) Rust v1.67 or later is required to build.
- (#825) `git submit` only fetches the necessary branches, rather than all branches, before pushing.
- (#826) Switch to `scm-record` crate to provide the partial commit interface.
- (#914) The default revset for `git restack` is now `draft()`.

### Fixed

- (#866) New commits created during `git rebase` are no longer kept if the rebase is aborted.
- (#866) `git test` no longer clobbers working copy changes.
- (#876) After `git amend`, detached branches previously pointing to the amended commit are now updated.
- (#915) Fixed certain situations of rebasing merge commits when all parents have also been rebased.
- (#920) Running `git sync` when the main branch is checked out no longer leaves a dirty index.
- (#938) Somehow fixed mysterious hang.

## New Contributors

Thanks to the following first-time contributors!

* @melko made their first contribution in https://github.com/arxanas/git-branchless/pull/829
* @jakejx made their first contribution in https://github.com/arxanas/git-branchless/pull/825
* @alerque made their first contribution in https://github.com/arxanas/git-branchless/pull/857
* @thoughtpolice made their first contribution in https://github.com/arxanas/git-branchless/pull/978

If you're interested in contributing, check out the Developer Guide in the [Wiki](https://github.com/arxanas/git-branchless/wiki), or post in the [Discussions](https://github.com/arxanas/git-branchless/discussions) or [Discord server](https://discord.gg/caYQBJ82A4) to ask any questions.

<hr /><em>This discussion was created from the release <a href='https://github.com/arxanas/git-branchless/releases/tag/v0.8.0'>v0.8.0</a>.</em>

**Comment by tech189** [comment]
Looks like pre-built binaries are missing from this release and the action is failing: https://github.com/arxanas/git-branchless/actions/runs/6040012495

**Reply by arxanas** [reply]
I built the release binaries and manually attached them, thanks to @adzenith's fix in https://github.com/arxanas/git-branchless/pull/1060.
